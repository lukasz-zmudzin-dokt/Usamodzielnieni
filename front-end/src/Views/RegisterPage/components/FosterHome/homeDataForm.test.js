import {MemoryRouter} from "react-router-dom";
import HomeDataForm from "./homeDataForm";
import {getByTestId, getByText, waitForElement, fireEvent, render} from "@testing-library/react";
import React from "react";


describe('HomeDataForm', () => {
    let props;

    beforeAll(() => {
        props = {
            data: "",
            onBlur: jest.fn()
        }
    });

    it('should render correctly', () => {
        const  {container}  = render(
            <MemoryRouter>
                <HomeDataForm data={props.data} onBlur={props.onBlur} />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
    it('should call onBlur with name of home value', () => {
        const { getByPlaceholderText } = render(
            <MemoryRouter>
                <HomeDataForm data={props.data} onBlur={props.onBlur} />
            </MemoryRouter>
        );
        const input = getByPlaceholderText("Nazwa placówki");
        fireEvent.change(input, {target: {value: "Dom dziecka"}});
        expect(props.onBlur).toHaveBeenCalledWith(...props.data, {name_of_place: "Dom dziecka"});
    });

    it('should call onBlur with street value', () => {
        const { getByPlaceholderText } = render(
            <MemoryRouter>
                <HomeDataForm data={props.data} onBlur={props.onBlur} />
            </MemoryRouter>
        );
        const input = getByPlaceholderText("Ulica");
        fireEvent.change(input, {target: {value: "Poznańska"}});
        expect(props.onBlur).toHaveBeenCalledWith(...props.data, {street: "Poznańska"});
    });

    it('should call onBlur with city value', () => {
        const { getByPlaceholderText } = render(
            <MemoryRouter>
                <HomeDataForm data={props.data} onBlur={props.onBlur} />
            </MemoryRouter>
        );
        const input = getByPlaceholderText("Nazwa miasta");
        fireEvent.change(input, {target: {value: "Warszawa"}});
        expect(props.onBlur).toHaveBeenCalledWith(...props.data, {city: "Warszawa"});
    });

    it('should call onBlur with post code value', () => {
        const { getByPlaceholderText } = render(
            <MemoryRouter>
                <HomeDataForm data={props.data} onBlur={props.onBlur} />
            </MemoryRouter>
        );
        const input = getByPlaceholderText("Kod pocztowy");
        fireEvent.change(input, {target: {value: "01-234"}});
        expect(props.onBlur).toHaveBeenCalledWith(...props.data, {city_code: "01-234"});
    });
});