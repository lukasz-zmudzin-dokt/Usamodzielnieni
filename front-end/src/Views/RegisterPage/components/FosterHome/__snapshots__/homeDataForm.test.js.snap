// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDataForm should render correctly 1`] = `
<div>
  <div
    class="loginPage__cardSection px-0 col-lg mr-lg-3 card bg-light"
  >
    <h4
      class="card-header"
    >
      Dane placówki
    </h4>
    <div
      class="account_form_card card-body"
    >
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupNameOfPlace"
          maxlength="60"
          minlength="1"
          name="name_of_place"
          placeholder="Nazwa placówki"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj nazwę placówki
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupStreet"
          maxlength="120"
          minlength="1"
          name="street"
          placeholder="Ulica"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj ulicę na której znajduje się placówka
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupNumbah"
          maxlength="20"
          minlength="1"
          name="number"
          pattern="^([0-9]{1})[\\\\s\\\\S]*"
          placeholder="Numer budynku"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj numer budynku, przy którym znajduje się placówka
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupCity"
          maxlength="40"
          minlength="1"
          name="city"
          placeholder="Nazwa miasta"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj nazwę miasta
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupCityCode"
          maxlength="6"
          minlength="1"
          name="city_code"
          pattern="[0-9]{2}[-][0-9]{3}"
          placeholder="Kod pocztowy"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj kod pocztowy w formacie 00-000
        </div>
      </div>
    </div>
  </div>
</div>
`;
